build-logic/src/main/java/qsl/internal/extension/QslModuleExtensionImpl.java

```diff
@@ -0,0 +1,180 @@
+package qsl.internal.extension;
+
+import java.io.Serial;
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import net.fabricmc.loom.api.LoomGradleExtensionAPI;
+import org.gradle.api.Action;
+import org.gradle.api.Named;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.Project;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Nested;
+import qsl.internal.dependency.QslLibraryDependency;
+import qsl.internal.json.Environment;
+import qsl.internal.license.LicenseHeader;
+import qsl.internal.task.ApplyLicenseTask;
+import qsl.internal.task.CheckLicenseTask;
+
+public class QslModuleExtensionImpl extends QslExtension implements Serializable {
+	@Serial
+	// increment when changing this class to properly invalidate the generateQmj task
+	private static final long serialVersionUID = 2L;
```

Comment:
Should probably be `1L` for now, and updated when changes are made to this file since the PR isn't merged yet